d2renderers/d2svg/d2svg.go

```diff
@@ -549,9 +549,18 @@ func drawConnection(writer io.Writer, labelMaskID string, connection d2target.Co
 		markerEnd = fmt.Sprintf(`marker-end="url(#%s)" `, id)
 	}
 
+	srcAdj, dstAdj := getArrowheadAdjustments(connection, idToShape)
+	path := pathData(connection, srcAdj, dstAdj)
+	mask := fmt.Sprintf(`mask="url(#%s)"`, labelMaskID)
+
 	var labelTL *geo.Point
 	if connection.Label != "" {
-		labelTL = connection.GetLabelTopLeft()
+		srcArrowWidth, _ := arrowheadDimensions(connection.SrcArrow, float64(connection.StrokeWidth))
+		dstArrowWidth, _ := arrowheadDimensions(connection.DstArrow, float64(connection.StrokeWidth))
+
+		srcAdjDistance := geo.EuclideanDistance(srcAdj.X, srcAdj.Y, 0, 0) + srcArrowWidth
+		dstAdjDistance := geo.EuclideanDistance(dstAdj.X, dstAdj.Y, 0, 0) + dstArrowWidth
+		labelTL = connection.GetLabelTopLeft(srcAdjDistance, dstAdjDistance)
```

Comment:
instead of updating `GetLabelTopLeft`, and `GetPointOnRoute`, I think it might be better to temporarily move the route start and end points to begin after the arrowheads (e.g. `connection.Route[0].X += srcAdj.X`) and then revert this after getting the label position on the route between arrowheads. This avoids the extra complexity in the `GetPointOnRoute` funcs.

Reply:
to temporarily construct the route, we can't just add a start and end point to original route, but also we need to remove  the extra points before and after, can this be done by calc the distance along the route? what if the connection is a curve?

Reply:
what if there are multiple points between original point and new point? if so there would still be issue

Reply:
that's true but there shouldn't be multiple points within the length of the arrowhead. It will be more robust to account for that though in case it does come up.